cvat-ui/src/components/common/cvat-table.tsx

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Row, Col } from 'antd/lib/grid';
+import Table, { TableProps } from 'antd/lib/table';
+import Input from 'antd/lib/input';
+import Text from 'antd/lib/typography/Text';
+import Button from 'antd/lib/button';
+import Space from 'antd/lib/space';
+import Popover from 'antd/lib/popover';
+import Checkbox, { CheckboxChangeEvent } from 'antd/lib/checkbox';
+import { DownloadOutlined, MoreOutlined } from '@ant-design/icons';
+import { Config } from '@react-awesome-query-builder/antd';
+import jsonLogic from 'json-logic-js';
+
+import { ResourceFilterHOC, defaultVisibility } from 'components/resource-sorting-filtering';
+
+type Props = TableProps & {
+    onFilterDataSource?(data: TableProps['dataSource']): void;
+    onChangeColumnVisibility?(id: number, isHidden: boolean): void;
+    renderExtraActions?(): JSX.Element;
+    queryBuilderConfig?: Partial<Config>;
+    queryBuilderPredefinedQueries?: Record<string, string>;
+    queryBuilderCache?: {
+        key: string;
+        capacity: number;
+    };
+    csvExport?: {
+        filename: string;
+    };
+    tableTitle?: string;
+    searchDataIndex?: (string | string[])[];
+};
+
+/*
+    Enhanced table component
+    Supports:
+    * titles
+    * advanced filtration with json query
+    * overall search
+    * csv export
+    * show/hide columns
+    Current restrictions:
+    * queryBuilderConfig, queryBuilderPredefinedQueries, queryBuilderCache are supposed to be static
+*/
+function CVATTable(props: Props): JSX.Element {
+    const {
+        onChangeColumnVisibility,
+        onFilterDataSource,
+        renderExtraActions,
+        queryBuilderConfig,
+        queryBuilderCache,
+        queryBuilderPredefinedQueries,
+        searchDataIndex,
+        tableTitle,
+        dataSource,
+        csvExport,
+        columns,
+        ...rest
+    } = props;
+
+    const [FilteringComponent, setFilteringComponent] = useState<ReturnType<typeof ResourceFilterHOC> | null>(null);
+    const [filterValue, setFilterValue] = useState<string | null>(null);
+    const [searchPhrase, setSearchPhrase] = useState<string | null>(null);
+    const [visibility, setVisibility] = useState(defaultVisibility);
+    const [filteredDataSource, setFilteredDataSource] = useState<typeof dataSource>(dataSource);
+    const [modifiedColumns, setModifiedColumns] = useState<typeof columns>([]);
+
+    const downloadCSV = useCallback(() => {
```

Comment:
I think we should add a way to manipulate csvExport externally such as: rename existing column, add new colums for data rows.
It can be useful to: 1. Add a column with a link to frame/job (which is now combined in 1 column in column with ID) 2. Rename 'Actions' to 'Active' in management table for more clarity

Reply:
> I think we should add a way to manipulate csvExport externally such as: rename existing column, add new colums for data rows.

The component may be enhanced by demand
Now there is not need to implement it as it covers existing features.

Reply:
Without it we will lose the correct column name('Actions` -> `Active`) for management page table